torch/testing/_internal/common_device_type.py

```diff
@@ -783,12 +800,24 @@ def only_fn(self, device, *args, **kwargs):
 # precisions (or are working with multiple dtypes) they should be specified
 # explicitly and computed using self.precision (e.g.
 # self.precision *2, max(1, self.precision)).
+#
+# In addition, rel_precision is added to set the relative precision to the
+# corresponding override.
+# Ex.
+#
+# precisionOverride({torch.half : 1e-2, torch.float : 1e-4,
```

Comment:
I appreciate this approach and it's great that this PR identified the need for a means for setting relative precision. I also think the mechanism applying this change looks good. As @kshitij12345's comments suggest, however, this UX seems a little shoehorned in. How about we consider some alternatives?

One option would be to add a new decorator, `relativeTolerance`, which acts like the current `precisionOverride` decorator but only sets relative tolerance. That seems a little limited if people want to set both atol and rtol, though.

Second option would be to create a different decorator, `toleranceOverride`, that could accept a dict whose keys were dtypes and whose values were dicts or named tuples with two values (for now): atol and rtol. Then it would override both atol and rtol. To be consistent with assertEqual would could even require that both atol and rtol be set.

What would you think of these approaches, @wqwu-fb? The implementation of the decorator could even be separated into a different PR. We could replace a use of the current `precisionOverride` decorator with it to test it.

A more general `toleranceOverride` should also probably replace the existing `precisionOverride` decorator, so we should add some comments in the code reflecting that.

Reply:
I feel like toleranceOverride is better.